lib/iris/tests/unit/common/mixin/test_LimitedAttributeDict.py

```diff
@@ -42,11 +42,24 @@ def test___eq___numpy(self):
         right = LimitedAttributeDict(**values)
         self.assertEqual(left, right)
         self.assertEqual(left, values)
-        values = dict(one=np.arange(1), two=np.arange(1), three=np.arange(1))
-        left = LimitedAttributeDict(dict(one=0, two=0, three=0))
-        right = LimitedAttributeDict(**values)
-        self.assertEqual(left, right)
-        self.assertEqual(left, values)
+
+        # Test inequality:
+        values = dict(one=np.arange(1), two=np.arange(2), three=np.arange(3))
+        left = LimitedAttributeDict(**values)
+        right = LimitedAttributeDict(
+            one=np.arange(3), two=np.arange(2), three=np.arange(1)
+        )
+        self.assertNotEqual(left, right)
+        self.assertNotEqual(values, right)
+
+        # ChrisB: I am not sure the follow test is really valid. It is comparing
+        # a one-element numpy array with a scalar. Is that what we want?
+
+    #    values = dict(one=np.arange(1), two=np.arange(1), three=np.arange(1))
+    #    left = LimitedAttributeDict(dict(one=0, two=0, three=0))
+    #    right = LimitedAttributeDict(**values)
+    #    self.assertEqual(left, right)
+    #    self.assertEqual(left, values)
```

Comment:
I think this usage **is** intentional.
We do want x to be the same as [x], because that's how netcdf attributes work.
So 1 = [1], but (obviously) 1 != [1, 0].

I think at least one way to handle this would be to wrap both sides with np.array(x, ndmin=1).
Interestingly, that leaves open the question of whether attributes can be multidimensional, or indeed non-array objects.  Iris does allow that, but they can't exist in files:   We use this. for example, in allowing attributes to be [STASH objects](https://github.com/SciTools/iris/blob/cb475400c471f79fa129188fae9fd97140156ea2/lib/iris/fileformats/pp.py#L246) -- but these are specifically intercepted + converted (both ways) on netcdf read and write.